### PR TITLE
Remove 'asDiagram' infrastructure

### PR DESCRIPTION
**Pull request by niklas-wortmann**
This cleans up the old way of generating marble diagrams. It also hides the internal testing document from the docs, we can still refer to that but it's not visible from the menu anymore.

That's how the new diagrams (inlined as svg) do kinda look like:
image

### CODE REVIEW

**Comment by timdp** [comment]
Guessing this relates to #5019, #5100, and #5402, so just cross-linking for visibility...

File: marbles/scripts/index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { renderMarbleDiagram } from 'swirly-renderer-node';
+import { readdir, readFileSync, writeFileSync } from 'fs';
+import { join } from 'path';
+import { parseMarbleDiagramSpec } from 'swirly-parser';
+import { styles as defaultLight } from 'swirly-theme-default-light';
```
Comment (timdp) [inline marbles/scripts/index.ts:5]: It's fine if you don't provide all styles. Swirly will extend the default light theme with whatever you do provide.

File: marbles/scripts/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+};
+
+const styles: DiagramStyles = {
+    ...defaultLight,
+    frame_width: 20,
+    completion_height: 20,
+    higher_order_angle: 30,
+    arrow_fill_color: 'black',
+    background_color: 'rgba(255, 255, 255, 0.0)',
+    operator_fill_color: 'rgba(255, 255, 255, 0.0)'
```
Comment (timdp) [inline marbles/scripts/index.ts:43]: These two already default to white.

File: marbles/scripts/index.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const renderMarble = (path: string, fileName: string): Promise<true> => {
+    const file = readFileSync(join(path, fileName));
+    const diagramSpec = parseMarbleDiagramSpec(file.toString());
+    const { xml: unoptXml, width, height } = renderMarbleDiagram(diagramSpec, { styles });
+    const optimizedSVGPromise = optimizeXml(unoptXml);
+    return optimizedSVGPromise.then((svgXML) => {
+        const svgFileName = fileName.split('.')[0] + '.svg';
+        const svgPath = join(process.cwd(), 'docs_app', 'assets', 'images', 'svgs', svgFileName);
+        writeFileSync(svgPath, svgXML, { encoding: 'utf-8', flag: 'w' });
+        return true;
```
Comment (timdp) [inline marbles/scripts/index.ts:26]: I don't think this does anything.

**Comment by timdp** [comment]
Note that Swirly currently introduces a pseudo-circular dependency. In order to parse the ASCII marble diagrams, it depends on the stable `rxjs` to use `TestScheduler.parseMarbles()`. I assume that's not even a public API? Maybe it'd be good to extract it into a separate package, so that both `rxjs` and `swirly` can depend on it?

File: marbles/scripts/index.ts (hunk 1 above)
Comment (niklas-wortmann) [reply marbles/scripts/index.ts:5]: oh that's good to know. Thanks

File: marbles/scripts/index.ts (hunk 2 above)
Comment (niklas-wortmann) [reply marbles/scripts/index.ts:43]: I use them to make the svg transparent. The color doesn't really matter

File: marbles/scripts/index.ts (hunk 3 above)
Comment (niklas-wortmann) [reply marbles/scripts/index.ts:26]: yeah I prefered returning something instead of void, but the value is not really used, that's true

**Comment by niklas-wortmann** [comment]
Good call regarding the circular dependency. I don't know about the `TestScheduler.parseMarbles()` API, but I think as quick fix I could just move the marble diagram generation over to the `docs_app`.

**Comment by niklas-wortmann** [comment]
https://github.com/timdp/swirly/issues/2 Is kinda related to this pr as the generated SVGs could be improved in terms of accessibility

File: marbles/scripts/index.ts (hunk 2 above)
Comment (timdp) [reply marbles/scripts/index.ts]: Oh, sorry, I missed the alpha channel. SVG actually accepts `transparent` as a color, I think.

**Comment by timdp** [comment]
I've also introduced a workaround for the circular-ish dependency. As of v0.13.6, instead of depending on RxJS directly, the parser now uses a tree-shaken copy of `TreeScheduler.parseMarbles()`. So it's a rudimentary way of extracting the logic into a minimal package, at the expense of duplicating it.

**Comment by benlesh** [comment]
LGTM... @niklas-wortmann ... I'll let you merge this one. You probably want to cherry pick it over to 6.x as well, so I recommend doing a squash and merge.

**Comment by timdp** [comment]
Little bug: in package.json, you're depending on the `swirly` package directly, and pulling in all its dependencies (including Puppeteer) as a result.

**Comment by niklas-wortmann** [comment]
Oh I see, good call. Thanks yeah now that makes sense. I will fix that up

**Comment by niklas-wortmann** [comment]
@benlesh I don't think I will cherry pick it in V6. There's still a lot of work needed for that and I don't want to deal with the hassle :D Also it makes the V7 release even better having lots of great docs stuff too :D

**Comment by timdp** [comment]
I'm a big fan of `depcheck` to avoid stuff like this. Could be something for a follow-up PR. :-)

**Pull request merged**